models/userModel.js

```diff
@@ -0,0 +1,40 @@
+import mongoose from "mongoose";
+
+const userSchema = new mongoose.Schema({
+    username: {
+        type: String,
+        required: true,
+        min: 4,
+        max: 255,
```

Comment:
Please remember that some databases saves always data with max value, not sure how mongo works but this is case for example in mssql, it is not a big deal but it is worth remember, this might be something you can mention in interview for better impression

Reply:
with required you can return custom error, what kind of error you get when name is null?